src/integrationtests/java/com/aws/greengrass/integrationtests/logmanager/LogManagerTest.java

```diff
@@ -350,4 +353,105 @@ void GIVEN_log_manager_in_errored_state_WHEN_restarted_THEN_logs_upload_is_reatt
+        when(cloudWatchLogsClient.putLogEvents(any(PutLogEventsRequest.class)))
+                .thenReturn(PutLogEventsResponse.builder().nextSequenceToken("nextToken").build());
+        logManagerService.ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG.set(true);
+        tempDirectoryPath = Files.createTempDirectory(tempRootDir, "IntegrationTestsTemporaryLogFiles");
+
+        for (int i = 0; i < 5; i++) {
+            createTempFileAndWriteData(tempDirectoryPath, "integTestRandomLogFiles.log_",  "");
+        }
+        createTempFileAndWriteData(tempDirectoryPath, "integTestRandomLogFiles.log",  "");
+
+        setupKernel(tempDirectoryPath, "smallPeriodicIntervalUserComponentConfig.yaml");
+        //TOdo: a better mechanism will be written
+        TimeUnit.SECONDS.sleep(30);
+        verify(cloudWatchLogsClient, atLeastOnce()).putLogEvents(captor.capture());
+
+        List<PutLogEventsRequest> putLogEventsRequests = captor.getAllValues();
+        assertEquals(1, putLogEventsRequests.size());
+        for (PutLogEventsRequest request : putLogEventsRequests) {
+            assertEquals(calculateLogStreamName(THING_NAME, LOCAL_DEPLOYMENT_GROUP_NAME), request.logStreamName());
+            assertEquals("/aws/greengrass/UserComponent/" + AWS_REGION + "/UserComponentA",
+                    request.logGroupName());
+            assertNotNull(request.logEvents());
+            assertEquals(DEFAULT_LOG_LINE_IN_FILE * 6, request.logEvents().size());
+            assertEquals(DEFAULT_FILE_SIZE * 6,
+                    request.logEvents().stream().mapToLong(value -> value.message().length())
+                    .sum());
+        }
+        File folder = tempDirectoryPath.toFile();
+        Pattern logFileNamePattern = Pattern.compile("^integTestRandomLogFiles.log\\w*");
+        List<File> allFiles = new ArrayList<>();
+        File[] files = folder.listFiles();
+        if (files != null) {
+            for (File file : files) {
+                if (file.isFile()
+                        && logFileNamePattern.matcher(file.getName()).find()
+                        && file.length() > 0) {
+                    allFiles.add(file);
+                }
+            }
+        }
+        assertEquals(1, allFiles.size());
+        logManagerService.ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG.set(false);
+    }
+
+    //TODO: this test is only for getting some certain level of knowledge of current change uploading active log file
+    // . It will be eventually removed.
+    @Test
+    void GIVEN_system_config_with_small_periodic_interval_WHEN_active_logs_included_THEN_logs_are_uploaded_to_cloud(
+            ExtensionContext ec) throws Exception {
+        when(cloudWatchLogsClient.putLogEvents(any(PutLogEventsRequest.class)))
+                .thenReturn(PutLogEventsResponse.builder().nextSequenceToken("nextToken").build());
+        logManagerService.ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG.set(true);
+        ignoreExceptionOfType(ec, NoSuchFileException.class);
```

Comment:
This boolean value is only for giving a chance for this small PR to perform some tests without changing the current logic, such that all tests can pass and this small PR can be merged. This is only for serving the purpose of avoiding massive PR, and this static boolean constant will be removed in the end.